activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/JobExecutorIT.java

```diff
@@ -0,0 +1,402 @@
+import org.activiti.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.activiti.engine.runtime.Job;
+import org.activiti.engine.runtime.ProcessInstance;
+import org.activiti.spring.SpringProcessEngineConfiguration;
+import org.activiti.spring.boot.ProcessEngineConfigurationConfigurer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.cloud.stream.binder.ConsumerProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@TestPropertySource("classpath:application-test.properties")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, properties = {
+        "activiti.cloud.rb.job-executor.enabled=true",
+        "activiti.cloud.rb.job-executor.message-job-consumer.max-attempts=4"
+})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+public class JobExecutorIT {
+    private static final Logger logger = LoggerFactory.getLogger(JobExecutorIT.class);
+
+    private static final String TEST_BOUNDARY_TIMER_EVENT = "testBoundaryTimerEvent";
+    private static final String START_TIMER_EVENT_EXAMPLE = "startTimerEventExample";
+    private static final String INTERMEDIATE_TIMER_EVENT_EXAMPLE = "intermediateTimerEventExample";
+
+    private static final String ASYNC_TASK = "asyncTask";
+
+    @Autowired
+    private RuntimeService runtimeService;
+    
+    @Autowired
+    private ManagementService managementService;
+
+    @Autowired
+    private RepositoryService repositoryService;
+
+    @Autowired
+    private ConsumerProperties messageJobConsumerProperties;
+
+    @Autowired
+    private MessageBasedJobManager messageBasedJobManager;
+    
+    @SpyBean
+    private JobMessageProducer jobMessageProducer;
+    
+    private ProcessEngineConfiguration processEngineConfiguration;
+    
+    @Autowired
+    private MessageHandler jobMessageHandler;
+
+    @TestConfiguration
+    static class JobExecutorITProcessEngineConfigurer implements ProcessEngineConfigurationConfigurer {
+        
+        @Override
+        public void configure(SpringProcessEngineConfiguration processEngineConfiguration) {
+            processEngineConfiguration.setAsyncExecutorDefaultTimerJobAcquireWaitTime(1000);
```

Comment:
Here, I think we can reduce even more the wait time to impact as less as possible the execution time for this test: `500` ms should be ok.

Reply:
Sounds good. I will try.

Reply:
Fixed